### PR TITLE
Updated glossary and embedded videos

### PR DESCRIPTION
**Pull request by mfateev**

### CODE REVIEW

File: docs/04_glossary.md [hunk 1]
```diff
@@ -33,15 +40,16 @@ The Go client doesn't use this.
 
 ### Decision
 Any action taken by the workflow durable function is called a decision. For example:
-scheduling an activity, canceling a child workflow, or starting a timer.
+scheduling an activity, canceling a child workflow, or starting a timer. A [decision task](completion) contains an optional list of decisions. Every decision is recorded in the [event history](#event-history) as an [event](#event)
```
Comment (dmetzgar) [inline docs/04_glossary.md]: `[decision task](completion)` -> `[decision task](#decision-task)`?

File: docs/04_glossary.md [hunk 2]
```diff
@@ -33,15 +40,16 @@ The Go client doesn't use this.
...
 
 ### Decision Task
-
-Every time a new external event that might affect a workflow state is recorded. a decision task
-notifies a workflow worker about it. After the new event is handled, the decision task is completed.
+Every time a new external event that might affect a workflow state is recorded a decision task that contains it is added to a [decision-task-list] and then picked up by a workflow worker. After the new event is handled, the decision task is completed with a list of [decisions](#decision)
```
Comment (dmetzgar) [inline docs/04_glossary.md]: `[decision-task-last]` -> `[decision task list](#decision-task-list)`

File: docs/04_glossary.md [hunk 3]
```diff
@@ -101,15 +105,15 @@ those tasks, and communicates task execution results back to the Cadence service
...
+A fault-oblivious stateful function that orchestrates activities. A *Workflow* has full control over
 which activities are executed, and in which order. A *Workflow* must not affect
 the external world directly, only through activities. What makes workflow code
 a *Workflow* is that its state is preserved by Cadence. Therefore any failure
 of a worker process that hosts the workflow code does not affect the workflow
 execution. The *Workflow* continues as if these failures did not happen. At the
 same time, activities can fail any moment for any reason. Because workflow code
-is fully fault tolerant, it is guaranteed to get notifications about activity
-failure or timeouts and act accordingly. There is no limit on potential workflow
+is fully fault-obliviouss, it is guaranteed to get notifications about activity
```
Comment (dmetzgar) [inline docs/04_glossary.md]: extra 's' in 'obliviouss'

File: docs/04_glossary.md (hunk 1 above)
Comment (purplesusan) [reply docs/04_glossary.md]: and please end the sentence with a period :)

File: docs/04_glossary.md (hunk 2 above)
Comment (purplesusan) [reply docs/04_glossary.md]: A comma after "is recorded" would be nice.
same comment about ending with a period.

**Pull request merged**